src/number_theory/number_field/embeddings.lean

```diff
@@ -123,3 +125,134 @@ end
 end bounded
 
 end number_field.embeddings
+
+section place
+
+variables {K : Type*} [field K] {A : Type*} [normed_division_ring A] [nontrivial A] (φ : K →+* A)
+
+/-- An embedding into a normed division ring defines a place of `K` -/
+def number_field.place :
+  absolute_value K ℝ := absolute_value.comp (is_absolute_value.to_absolute_value (norm : A → ℝ))
+    φ.injective
+
+lemma number_field.place_apply (x : K) :
+  (number_field.place φ) x = norm(φ x) := by refl
+
+end place
+
+namespace number_field.complex_embedding
+
+open complex number_field
+
+open_locale complex_conjugate
+
+variables {K : Type*} [field K]
+
+/-- The conjugate of a complex embedding as a complex embedding. -/
+def conjugate (φ : K →+* ℂ) : K →+* ℂ := star φ
+
+@[simp]
+lemma conjugate_coe_eq (φ : K →+* ℂ) (x : K) : (conjugate φ) x = conj (φ x) := rfl
+
+lemma conjugate_conjugate_eq (φ : K →+* ℂ) :
+  conjugate (conjugate φ) = φ := star_involutive φ
+
+lemma place_conjugate_eq_place (φ : K →+* ℂ) (x : K) : place (conjugate φ) x = place φ x :=
+by simp only [place_apply, norm_eq_abs, abs_conj, conjugate_coe_eq]
+
+/-- A embedding into `ℂ` is real if it is fixed by complex conjugation. -/
+def is_real (φ : K →+* ℂ) : Prop := is_self_adjoint φ
+
+lemma is_real_iff {φ : K →+* ℂ} : is_real φ ↔ conjugate φ = φ := is_self_adjoint_iff
+
+/-- A real embedding as a ring homomorphism from `K` to `ℝ` . -/
+def is_real.embedding {φ : K →+* ℂ} (hφ : is_real φ) : K →+* ℝ :=
+{ to_fun := λ x, (φ x).re,
+  map_one' := by simp only [map_one, one_re],
+  map_mul' := by simp only [complex.eq_conj_iff_im.mp (ring_hom.congr_fun hφ _), map_mul, mul_re,
+  mul_zero, tsub_zero, eq_self_iff_true, forall_const],
+  map_zero' := by simp only [map_zero, zero_re],
+  map_add' := by simp only [map_add, add_re, eq_self_iff_true, forall_const], }
```

Comment:
I think this is a slightly more bundled version of `self_adjoint_part`? If so, it would be good to show that they are equal.

Reply:
Well, I don't know, the main point of this definition is that the image is in $\mathbb{R}$ whereas `self_adjoint_part` would give  a map with image into ℂ, I think. What we could prove is that `is_real.embedding` composed with `coe : ℝ → ℂ` is indeed `self_adjoint_part` of $\phi$. But to state the lemma already requires some work: need to prove `add_comm_group (K →+* ℂ)`, `star_add_monoid (K →+* ℂ)`, ...

Reply:
Ah, good point; I think you can ignore my suggestion here